src/Discord/Internal/Types/Guild.hs

```diff
@@ -119,26 +130,42 @@ instance FromJSON GuildUnavailable where
        GuildUnavailable <$> o .: "id"
 
 data PresenceInfo = PresenceInfo
-  { presenceUserId     :: UserId
+  { presenceUserId     :: Maybe UserId
   -- , presenceRoles   :: [RoleId]
-  , presenceActivities :: Maybe [Activity]
+  -- | Activities and the names of their buttons. The buttons field of Activity
+  -- will be blank, as the additional maybe field will have the button names it
+  -- would contain.
+  , presenceActivities :: Maybe [(Activity, Maybe [T.Text])]
   , presenceGuildId    :: Maybe GuildId
-  , presenceStatus     :: T.Text
+  , presenceStatus     :: Maybe T.Text
   } deriving (Show, Read, Eq, Ord)
 
 instance FromJSON PresenceInfo where
   parseJSON = withObject "PresenceInfo" $ \o ->
-    PresenceInfo <$> (o .: "user" >>= (.: "id"))
-                 <*> o .:  "activities"
+    PresenceInfo <$> (o .:? "user" >>= mapM (.: "id"))
+                 <*> (o .:? "activities" >>= parseActivities)
                  <*> o .:? "guild_id"
-                 <*> o .:  "status"
+                 <*> o .:? "status"
+    where
+    parseActivities :: Maybe [Value] -> Parser (Maybe [(Activity, Maybe [T.Text])])
+    parseActivities = \case
+      Nothing -> pure Nothing
+      Just vs -> Just <$> mapM parseIncomingActivity vs
+    parseIncomingActivity :: Value -> Parser (Activity, Maybe [T.Text])
+    parseIncomingActivity = withObject "PI Activity w/ BtnNames" $ \o -> do
+      let o' = KM.delete "buttons" o
+      act <- parseJSON (Object o')
+      buttonNames <- o .:? "buttons"
+      pure (act, buttonNames)
 
 -- | Object for a single activity
 --
 -- https://discord.com/developers/docs/topics/gateway#activity-object
```

Comment:
Minor note: the docs changed and this url should be changed to `https://discord.com/developers/docs/topics/gateway-events#activity-object`